src/main.ts

```diff
@@ -330,10 +412,117 @@ class NixInstallerAction {
 
     // Normal just doing of the install
     const binary_path = await this.fetch_binary();
+    actions_core.startGroup("Installing Nix");
     await this.execute_install(binary_path);
+    actions_core.endGroup();
+
+    if (this.force_docker_shim) {
+      await this.spawnDockerShim();
+    }
     await this.set_github_path();
   }
 
+  async spawnDockerShim(): Promise<void> {
+    actions_core.startGroup(
+      "Configuring the Docker shim as the Nix Daemon's process supervisor",
+    );
+
+    const images: { [key: string]: string } = {
+      X64: path.join(__dirname, "/../docker-shim/amd64.tar.gz"),
+      ARM64: path.join(__dirname, "/../docker-shim/arm64.tar.gz"),
+    };
+
+    let arch;
+    if (process.env.RUNNER_ARCH === "X64") {
+      arch = "X64";
+    } else if (process.env.RUNNER_ARCH === "ARM64") {
+      arch = "ARM64";
+    } else {
+      throw Error("Architecture not supported in Docker shim mode.");
+    }
+    actions_core.debug("Loading image: determinate-nix-shim:latest...");
+    {
+      const exit_code = await actions_exec.exec(
+        "docker",
+        ["image", "load", "--input", images[arch]],
+        {
+          listeners: {
+            stdout: (data: Buffer) => {
+              const trimmed = data.toString("utf-8").trimEnd();
+              if (trimmed.length >= 0) {
+                actions_core.debug(trimmed);
+              }
+            },
+            stderr: (data: Buffer) => {
+              const trimmed = data.toString("utf-8").trimEnd();
+              if (trimmed.length >= 0) {
+                actions_core.debug(trimmed);
+              }
+            },
+          },
+        },
+      );
+
+      if (exit_code !== 0) {
+        throw new Error(
+          `Failed to build the shim image, exit code: \`${exit_code}\``,
+        );
+      }
+    }
+
+    {
+      actions_core.debug("Starting the Nix daemon through Docker...");
+      const exit_code = await actions_exec.exec(
+        "docker",
+        [
+          "--log-level=debug",
+          "run",
```

Comment:
Do you want a `--restart always` or `--restart on-failure` maybe?

Reply:
Nice idea.